### PR TITLE
 fix: Add LegalBench datasets - 6

### PR DESCRIPTION
**Pull request by awinml**
## Checklist for adding MMTEB dataset

Reason for dataset addition: Showcases the model's performance on the [LegalBench](https://hazyresearch.stanford.edu/legalbench/) datasets, enabling evaluation on domain-specific legal texts.

Follow up to #613 

## Changes Made:

The following 8 datasets have been added in this PR:

1. [CUAD No Solicit of Customers](https://hazyresearch.stanford.edu/legalbench/tasks/cuad_no-solicit_of_customers.html)
2. [CUAD No Solicit of Employees](https://hazyresearch.stanford.edu/legalbench/tasks/cuad_no-solicit_of_employees.html)
3. [CUAD Non Compete](https://hazyresearch.stanford.edu/legalbench/tasks/cuad_non-compete.html)
4. [CUAD Non Disparagement](https://hazyresearch.stanford.edu/legalbench/tasks/cuad_non-disparagement.html)
5. [CUAD Non Transferable License](https://hazyresearch.stanford.edu/legalbench/tasks/cuad_non-transferable_license.html)
6. [CUAD Notice Period To Terminate Renewal](https://hazyresearch.stanford.edu/legalbench/tasks/cuad_notice_period_to_terminate_renewal.html)
7. [CUAD Post Termination Services](https://hazyresearch.stanford.edu/legalbench/tasks/cuad_post-termination_services.html)
8. [CUAD Price Restrictions](https://hazyresearch.stanford.edu/legalbench/tasks/cuad_price_restrictions.html)

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/eng/LegalBenchClassification.py [hunk 1]
```diff
@@ -1512,3 +1512,435 @@ def dataset_transform(self):
...
+            "path": "nguha/legalbench",
+            "name": "cuad_non-transferable_license",
+            "revision": "12ca3b695563788fead87a982ad1a068284413f4",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=("2000-01-01", "2023-08-23"),
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/eng/LegalBenchClassification.py:2180]:
```suggestion
        date=("2000-01-01", "2023-08-23"), # best guess
```
Assuming it is only an approximate guess

Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/eng/LegalBenchClassification.py:2180]: would redo for the other datasets where this is the case as well

File: docs/mmteb/points/622.jsonl [hunk 2]
```diff
@@ -0,0 +1 @@
+{"GitHub": "awinml", "New dataset": 16}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/622.jsonl:1]: add points for review

**Review by KennethEnevoldsen — APPROVED** [review]
Only one minor comment, but otherwise it looks good

File: mteb/tasks/Classification/eng/LegalBenchClassification.py (hunk 1 above)
Comment (awinml) [reply mteb/tasks/Classification/eng/LegalBenchClassification.py:2180]: Their code explicitly mentions the years, please see: https://github.com/TheAtticusProject/cuad/blob/main/scrape.py#L16, so I don't feel it is necessary here. But I'm open to adding it if you feel strongly about it.

Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/eng/LegalBenchClassification.py:2180]: Then I would def. not add the comment. Thanks for the clarification

**Comment by KennethEnevoldsen** [comment]
Everything looks good - Enabled auto-merge

**Pull request merged**